panimg/image_builders/dicom.py

```diff
@@ -363,35 +363,49 @@ def _get_headers_by_study(
     A dictionary of sorted headers for all dicom image files found within path,
     grouped by study id.
     """
-    studies: Dict[str, Dict[str, Any]] = {}
-    indices: Dict[str, Dict[str, int]] = {}
+    study_key = Tuple[Any, Any, Any]
```

Comment:
This could also be `Tuple[str, str, str]` actually, the header values will always be strings.